omegaconf/basecontainer.py

```diff
@@ -368,6 +389,10 @@ def expand(node: Container) -> None:
         if src_type is not None and not is_primitive_dict(src_type):
             dest._metadata.object_type = src_type
 
+        if dest._metadata.ref_type is Any:
+            dest._metadata.ref_type = src_ref_type
+            dest._metadata.optional = src._metadata.optional
+
```

Comment:
I think we should only do this when we are promoting the type of dest to the type of src. (the case of missing overriding non-missing).

Reply:
Changed in 234067894abfd761ff754dfcc827d6fdfc32110d

At the same time this commit fixed an issue where `src_ref_type` could be `Optional[X]`, but we want to get rid of the optional part when assigning `_metadata.ref_type`.